server/graphql/invoice/src/mutations/inbound_shipment/add_from_master_list.rs

```diff
@@ -0,0 +1,311 @@
+use crate::mutations::{
+    AddFromMasterListResponse, AddToShipmentFromMasterListInput, DeleteError, DeleteErrorInterface,
+    MasterListNotFoundForThisName,
+};
+use async_graphql::*;
+use graphql_core::{
+    simple_generic_errors::{CannotEditInvoice, RecordNotFound},
+    standard_graphql_error::validate_auth,
+    standard_graphql_error::StandardGraphqlError,
+    ContextExt,
+};
+use graphql_types::types::InvoiceLineConnector;
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    invoice::common::AddToShipmentFromMasterListError as ServiceError,
+};
+
+pub fn add_from_master_list(
+    ctx: &Context<'_>,
+    store_id: &str,
+    input: AddToShipmentFromMasterListInput,
+) -> Result<AddFromMasterListResponse> {
+    validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateOutboundShipment,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context()?;
+
+    let response = match service_provider
+        .invoice_service
+        .add_to_inbound_shipment_from_master_list(&service_context, store_id, input.to_domain())
+    {
+        Ok(invoice_lines) => {
+            AddFromMasterListResponse::Response(InvoiceLineConnector::from_vec(invoice_lines))
+        }
+        Err(error) => AddFromMasterListResponse::Error(DeleteError {
+            error: map_error(error)?,
+        }),
+    };
+
+    Ok(response)
+}
+
+fn map_error(error: ServiceError) -> Result<DeleteErrorInterface> {
+    use StandardGraphqlError::*;
+    let formatted_error = format!("{:#?}", error);
+
+    let graphql_error = match error {
+        // Structured Errors
+        ServiceError::ShipmentDoesNotExist => {
+            return Ok(DeleteErrorInterface::RecordNotFound(RecordNotFound {}))
+        }
+        ServiceError::CannotEditShipment => {
+            return Ok(DeleteErrorInterface::CannotEditInvoice(
+                CannotEditInvoice {},
+            ))
+        }
+        ServiceError::MasterListNotFoundForThisName => {
+            return Ok(DeleteErrorInterface::MasterListNotFoundForThisName(
+                MasterListNotFoundForThisName {},
+            ))
+        }
+        // Standard Graphql Errors
+        ServiceError::NotThisStoreShipment => BadUserInput(formatted_error),
+        ServiceError::NotAnOutboundShipment => BadUserInput(formatted_error),
+        ServiceError::DatabaseError(_) => InternalError(formatted_error),
+        _ => BadUserInput(formatted_error),
```

Comment:
I don't expect an error of 'NotAnOutboundShipment' here.. have used a common error type since they are almost identical